src/bindings/event.ts

```diff
@@ -12,9 +12,6 @@ import { validateConfigChange } from "../handlers/push";
 let botContext: Context = {} as Context;
```

Comment:
yes, that is what I was thinking but there are alot of functions. I just wanted to confirm once before making that big of a change as I thought there might be a better way to do it. 
I'll do the changes as required
Thanks, for your review!

Reply:
Hey @whilefoo, am I going in the right direction?

Reply:
yes, just make sure context is always the first argument

Reply:
Hi @whilefoo , I did as asked for almost all the files however there are some errors that I'm not able to resolve that easily 
1. src/handlers/comment/handlers/index.ts

`Type '(context: BotContext, body: string) => Promise<string | undefined>' is not assignable to type 'CommandsHandler'.`

2. src/handlers/processors.ts

`Type '(context: BotContext) => Promise<void>' is not assignable to type 'ActionHandler'.
 Types of parameters 'context' and 'args' are incompatible.
 Type 'string | undefined' is not assignable to type 'BotContext'.
 Type 'undefined' is not assignable to type 'BotContext'.`

3. src/index.ts

`Argument of type '(context: BotContext) => Promise<void>' is not assignable to parameter of type 'HandlerFunction<"workflow_run" | "branch_protection_rule" | "branch_protection_rule.created" | "branch_protection_rule.deleted" | "branch_protection_rule.edited" | "check_run" | "check_run.completed" | ... 232 more ... | "workflow_run.requested", SimplifiedObject>'. Types of parameters 'context' and 'event' are incompatible.`

any insights on how these errors can be resolved would be much appreiated

Reply:
1. This is because `bindEvents` receives probot's `Context` but you changed it to our custom `BotContext`.

2. You changed `run` function to include `BotContext` so it is expecting it when it called in `src/tests/summarize-test.ts` but I'm not sure why we even have that? @0xcodercrane

Reply:
It is originally coming from the telegram notification adapter which has plenty of parts that need to be removed/updated. 

but for now, I would say we should definitely remove it. It was something someone created for the test of the weekly posting on telegram.

Reply:
ok then @Sadaf-A you can remove `src/tests/summarize-test.ts` file

Reply:
> Sorry for the late reply
> 
>     1. This is because `bindEvents` receives probot's `Context` but you changed it to our custom `BotContext`.
> 
>     2. You changed `run` function to include `BotContext` so it is expecting it when it called in `src/tests/summarize-test.ts` but I'm not sure why we even have that? @0xcodercrane

This is because bindEvents receives probot's Context but you changed it to our custom BotContext.
I can't this of a solution to this. what do you think would be an optimal solution for this.

Reply:
in bindEvents you receive probot Context and you can create custom Context and then it's normally passed down

Reply:
> in bindEvents you receive probot Context and you can create custom Context and then it's normally passed down

can you add comments in the PR as I've already tried this and faced issues with TypeScript